File: ruby/raindrops/raindrops.rb
class Raindrops

  RULES = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }

  def self.convert(num)
    factors = RULES.keys.select { |f| num % f == 0 }
    return factors.empty? ? num.to_s : factors.map { |n| RULES[n] }.join
  end
end

module BookKeeping
  VERSION = 2
end

File: swift/raindrops/Sources/Raindrops.swift
struct Raindrops {
    let drops: Int

    init(_ drops: Int) {
        self.drops = drops
    }

    var sounds: String {
        var dropStr = ""
        if drops % 3 == 0 { dropStr += "Pling" }
        if drops % 5 == 0 { dropStr += "Plang" }
        if drops % 7 == 0 { dropStr += "Plong" }

        if dropStr.isEmpty { dropStr = "\(drops)" }
        return dropStr
    }
}
